src/Security/Domain/Authentication/Model/LocalProvider.php

```diff
@@ -256,4 +249,135 @@ public function getProviderRefreshToken(string $token): ?ProviderToken
     {
         return null;
     }
+
+    /**
+     * Check if local security policy is respected
+     *
+     * @param User $user
+     * @param SecurityPolicy $securityPolicy
+     * @param bool $doesPasswordMatch
+     */
+    private function respectLocalSecurityPolicyOrFail(
+        User $user,
+        SecurityPolicy $securityPolicy,
+        bool $doesPasswordMatch,
+    ): void {
+        $isUserBlocked = $this->isUserBlocked($user, $securityPolicy, $doesPasswordMatch);
+
+        $this->writeUserRepository->updateBlockingInformation($user);
+
+        if ($isUserBlocked) {
+            $this->info(
+                '[LOCAL PROVIDER] authentication failed because user is blocked',
+                [
+                    'contact_alias' => $user->getAlias(),
+                ],
+            );
+            throw AuthenticationException::userBlocked();
+        }
+
+        if ($doesPasswordMatch && $this->isPasswordExpired($user, $securityPolicy)) {
+            $this->info(
+                '[LOCAL PROVIDER] authentication failed because password is expired',
+                [
+                    'contact_alias' => $user->getAlias(),
+                ],
+            );
+            throw PasswordExpiredException::passwordIsExpired();
+        }
+    }
```

Comment:
```suggestion
         if ($securityPolicy->getPasswordExpirationDelay() !== null) {
                  if ($doesPasswordMatch && $this->isPasswordExpired($user, $securityPolicy)) {
                        $this->info(
                            '[LOCAL PROVIDER] authentication failed because password is expired',
                            [
                                'contact_alias' => $user->getAlias(),
                            ],
                        );
                        throw PasswordExpiredException::passwordIsExpired();
                    }
                }
        }

```
Only call this method if expirationDelay is set in policy